File: 프로그래머스/2/42584. 주식가격/주식가격.py
from collections import deque
import itertools

def solution(prices):
    d_prices = deque(prices)
    result = []
    
    while d_prices:
        price = d_prices.popleft()
        if d_prices:
            # if price <= min(d_prices):
            #     result.append( len(d_prices) )
            # else:
            count = 0
            for i in d_prices:
                if i >= price:
                    count += 1
                else:
                    count += 1
                    break
            result.append( count )
                
    result.append(0)
    return result

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const L = prices.length
    const answer = Array(L).fill(0)
    
    for(let i = 0; i < L; i++){
        let cnt = 0
        for(let j = i+1; j < L; j++){
            if(prices[i] <= prices[j]){
                cnt += 1
            }
            else{
                if( i !== L-1){
                    cnt += 1
                }
                break
            }
        }
        answer[i] = cnt
    }
    
    return answer;
}